File: snippets/cpp/VS_Snippets_CLR/Type_IsValueType/CPP/type_isvaluetype.cpp

// <Snippet1>
using namespace System;

// Declare an enum type.
public enum class NumEnum
{
   One, Two
};

int main()
{
    bool flag = false;
    NumEnum testEnum = NumEnum::One;
      
    // Get the type of testEnum.
    Type^ t = testEnum.GetType();
      
    // Get the IsValueType property of the testEnum
    // variable.
    flag = t->IsValueType;
    Console::WriteLine("{0} is a value type: {1}", t->FullName, flag);
}
// </Snippet1>

File: snippets/csharp/System/Type/IsValueType/type_isvaluetype.cs
// <Snippet1>
using System;

// Declare an enum type.
enum NumEnum { One, Two }

public class Example
{

    public static void Main(string[] args)
    {
        bool flag = false;
        NumEnum testEnum = NumEnum.One;
        // Get the type of testEnum.
        Type t = testEnum.GetType();
        // Get the IsValueType property of the testEnum variable.
        flag = t.IsValueType;
        Console.WriteLine($"{t.FullName} is a value type: {flag}");
    }
}
// The example displays the following output:
//        NumEnum is a value type: True
// </Snippet1>
